src/outagesim/backup_reliability.jl

```diff
@@ -396,59 +655,62 @@ function survival_with_battery(net_critical_load::Vector, starting_batt_soc_kwh:
 
 Return a dictionary of inputs required for backup reliability calculations. 
 
 # Arguments
 - `d::Dict`: REopt results dictionary. 
 - `r::Dict`: Dictionary of inputs for reliability calculations. If r not included then uses all defaults. values read from dictionary:
-    -gen_operational_availability::Real = 0.9998        Fraction of year generators not down for maintenance
-    -gen_failure_to_start::Real = 0.0066                Chance of generator starting given outage
-    -gen_failure_to_run::Real = 0.00157                 Chance of generator failing in each hour of outage
-    -num_gens::Int = 1                                  Number of generators. Will be determined by code if set to 0 and gen capacity > 0.1
-    -gen_capacity::Real = 0.0                           Backup generator capacity. Will be determined by REopt optimization if set less than 0.1
-    -num_battery_bins::Int = 100                        Internal value for modeling battery
-    -max_outage_duration::Int = 96                      Maximum outage hour modeled
-    -microgrid_only::Bool = false                       Determines how generator, PV, and battery act during islanded mode
+    -generator_operational_availability::Real = 0.9998      Fraction of year generators not down for maintenance
+    -generator_failure_to_start::Real = 0.0066              Chance of generator starting given outage
+    -generator_failure_to_run::Real = 0.00157               Chance of generator failing in each hour of outage
+    -num_generators::Int = 1                                Number of generators. Will be determined by code if set to 0 and gen capacity > 0.1
+    -generator_size_kw::Real = 0.0                          Backup generator capacity. Will be determined by REopt optimization if set less than 0.1
+    -num_battery_bins::Int = 101                            Internal value for discretely modeling battery state of charge
+    -max_outage_duration::Int = 96                          Maximum outage hour modeled
+    -microgrid_only::Bool = false                           Boolean to specify if only microgrid upgraded technologies run during grid outage 
 ```
 """
-function backup_reliability_inputs(d::Dict, p::REoptInputs; r::Dict = Dict())::Dict
+function backup_reliability_reopt_inputs(;d::Dict, p::REoptInputs, r::Dict = Dict())::Dict
     zero_array = zeros(length(p.time_steps))
+
+
     r2 = dictkeys_tosymbols(r)
-    critical_loads_kw = p.s.electric_load.critical_loads_kw
+    
+    net_critical_loads_kw = p.s.electric_load.critical_loads_kw
 
+    #TODO Change CHP to meet capacity not load
     if "CHP" in keys(d)
-        chp_generation =  get(d["CHP"], "year_one_electric_production_series_kw", zero_array)
-        critical_loads_kw .-= chp_generation
+        chp_generation =  get(d["CHP"], "size_kw", 0)
+        net_critical_loads_kw .-= chp_generation
     end
 
     microgrid_only = get(r, "microgrid_only", false)
```

Comment:
Need to handle multiple PVs. d["PV"] could be a list of dictionaries instead of a dictionary

Reply:
@hdunham and we would just aggregate/sum the PV production for the ERP? Alternatively to handling this within backup_reliability.jl, since we are not modeling multiple PV's explicitly in ERP, we could make this a requirement for the ERP **endpoint** instead of these functions - i.e. process the REopt inputs to handle/aggregate multiple PV's. What if the REopt outputs have multiple PVs but then the ERP user changes the PV size? How do we normalize the PV production to scale to different sizes? If this gets too tricky, I'd be fine only allowing one PV to be handled, such as the first one from REopt if there are multiple. We don't allow multiple PV's from the UI.